### DISCUSSION TITLE
Programmatic creation of projects

### DISCUSSION THREAD

**Discussion by charles-dawes** [riok/Kreya#180] [General]
Hello, 

Is there any way to programmatically create Kreya projects?  I am looking into a way to provide our developers with a bundle of a kreya project and the necessary proto files to complete gRPC requests.  I would rather programmatically generate these vs storing projects in a git repo and having to maintain that repo any time a proto changes.

**Comment by CommonGuy** [comment]
Currently, the only way to programmatically create a Kreya project would be to create the `krproj` file with the correct information, then opening Kreya to let it create the imported operations.

We will add this functionality to the Kreya CLI

**Reply by charles-dawes** [reply]
Do you have any documentation on what field are required in the kproj file in order to expedite our development effort and/or when the updates to the CLI will be available?

**Reply by CommonGuy** [reply]
The updates to the CLI may take a while...

The `krproj` file only needs an `id` field which should contain an unique GUID. For your use case, you also need to add an import stream to the project (in the `importStreams` field, which is an array). The easiest way would be to manually create a project and import your gRPC API definition, then see what the `krproj` file looks like. There aren't that many fields

**Reply by charles-dawes** [reply]
Awesome, thank you for the information, its much appreciated.

**Reply by CommonGuy** [reply]
The Kreya CLI now supports a basic command to create a project: https://kreya.app/docs/cli/commands/project/create/
